gcloud/datastore/query.py

```diff
@@ -347,10 +342,10 @@ class Iterator(object):
         datastore_pb.QueryResultBatch.MORE_RESULTS_AFTER_LIMIT,
     )
 
-    def __init__(self, query, connection, limit=None, offset=0,
+    def __init__(self, query, client, limit=None, offset=0,
```

Comment:
> Shouldn't there be an accompanying docstring update?

`Iterator` doesn't document arguments to `__init__` at all.  We can add that, if you think it is needed, given its "private" nature.

> Also, if the _client is bound to the query, shouldn't client be an optional argument?

`client` is passed through from the value passed to `fetch()`.

> I suppose it's moot as Iterator is effectively a non-public class.

Yup.